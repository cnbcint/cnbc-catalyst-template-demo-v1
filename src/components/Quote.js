import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";
import useAnimation from "../hooks/useAnimation";
import { breakpoints, getAnimation } from "../helpers/Layout";

const Container = styled.blockquote`
  font-size: 1.5rem;
  @media (min-width: ${breakpoints.mobile}) {
    font-size: 2rem;
  }
  line-height: 150%;
  background-color: transparent;
  padding: 0;
  margin: 0;
  small {
    font-size: 0.8rem;
    line-height: 1rem;
    margin-top: 1rem;
    display: block;
  }
  text-align: ${(props) => props.aligment};
  > hr {
    margin-top: 1rem;
    margin-bottom: 1rem;
    border-color: ${(props) =>
      props.dividerColor ? props.dividerColor + " !important" : ""};
  }
`;
const Copy = styled.div`
  font-style: italic;
`;
export default function Quote(props) {
  const { ref: refCopy } = useAnimation({
    animationName: getAnimation(props.aligment),
    duration: 500,
    delay: 0,
    loop: props.loopAnimations,
    autoStart: true,
  });
  const { ref: refDevider1 } = useAnimation({
    animationName: "fadeIn",
    duration: 500,
    delay: 125,
    loop: props.loopAnimations,
    autoStart: true,
  });
  const { ref: refDevider2 } = useAnimation({
    animationName: "fadeIn",
    duration: 500,
    delay: 125,
    loop: props.loopAnimations,
    autoStart: true,
  });
  return (
    <Container
      className={props.className}
      aligment={props.aligment}
      style={props.style}
      dividerColor={props.dividerColor}
    >
      {props.addDeviders && <hr ref={refDevider1} />}
      <Copy ref={refCopy}>{props.children}</Copy>
      {props.addDeviders && <hr ref={refDevider2} />}
    </Container>
  );
}

Quote.defaultProps = {
  loopAnimations: true,
  aligment: "center",
  addDeviders: true,
  className: "",
  style: {},
};

Quote.propTypes = {
  /**
   * Valid CSS color
   */
  dividerColor: PropTypes.string,
  /**
   * Show or hide deviders
   */
  addDeviders: PropTypes.bool,
  className: PropTypes.string,
  /**
  React CSS format to override rules has required by design
  */
  style: PropTypes.object,
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.array,
    PropTypes.element,
  ]).isRequired,
  /**
   Enable to reset animation when users scrolls back into the element.
  */
  loopAnimations: PropTypes.bool,
  /**
   Quote aligment this will also affect animation beeing used
  */
  aligment: PropTypes.oneOf(["left", "center", "right"]),
};
