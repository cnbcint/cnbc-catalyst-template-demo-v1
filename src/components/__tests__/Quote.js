import React from "react";
import Quote from "../Quote";
import { render } from "@testing-library/react";
import "jest-styled-components";

beforeAll(() => {
  window.IntersectionObserver = class {
    observe() {
      return null;
    }
    disconnect() {}
  };
});
describe("Quote", () => {
  it("Renders correctly", () => {
    const { container } = render(<Quote>Testing Content</Quote>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("Renders correctly with no deviders", () => {
    const { container } = render(
      <Quote addDeviders={false}>Testing Content</Quote>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("Renders the correct content", () => {
    const { container } = render(<Quote>Testing Content</Quote>);
    expect(container.firstChild.textContent).toMatch("Testing Content");
  });

  it("Passes the class name", () => {
    const { container } = render(
      <Quote className="testClass">Testing Content</Quote>
    );
    expect(container.firstChild.classList.contains("testClass")).toBe(true);
  });

  it("Set The right aligment", () => {
    const { container } = render(<Quote aligment="right">Testing Title</Quote>);
    expect(container.firstChild).toHaveStyleRule("text-align", "right");
  });
});

afterAll(() => {
  window.IntersectionObserver = null;
});
