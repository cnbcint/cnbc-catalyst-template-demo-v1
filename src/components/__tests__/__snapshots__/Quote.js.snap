// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote Renders correctly 1`] = `
.c0 {
  font-size: 1.5rem;
  line-height: 150%;
  background-color: transparent;
  padding: 0;
  margin: 0;
  text-align: center;
}

.c0 small {
  font-size: 0.8rem;
  line-height: 1rem;
  margin-top: 1rem;
  display: block;
}

.c0 > hr {
  margin-top: 1rem;
  margin-bottom: 1rem;
}

.c1 {
  font-style: italic;
}

@media (min-width:768px) {
  .c0 {
    font-size: 2rem;
  }
}

<blockquote
  class="c0"
>
  <hr
    class="pre-animated"
  />
  <div
    class="c1 pre-animated"
  >
    Testing Content
  </div>
  <hr
    class="pre-animated"
  />
</blockquote>
`;

exports[`Quote Renders correctly with no deviders 1`] = `
.c0 {
  font-size: 1.5rem;
  line-height: 150%;
  background-color: transparent;
  padding: 0;
  margin: 0;
  text-align: center;
}

.c0 small {
  font-size: 0.8rem;
  line-height: 1rem;
  margin-top: 1rem;
  display: block;
}

.c0 > hr {
  margin-top: 1rem;
  margin-bottom: 1rem;
}

.c1 {
  font-style: italic;
}

@media (min-width:768px) {
  .c0 {
    font-size: 2rem;
  }
}

<blockquote
  class="c0"
>
  <div
    class="c1 pre-animated"
  >
    Testing Content
  </div>
</blockquote>
`;
